manifests/cluster-install/argo-server-rbac/argo-server-clusterole.yaml

```diff
@@ -29,6 +29,7 @@ rules:
       - list
       - watch
       - delete
+      - patch
```

Comment:
this is a pretty minor permission given the Server already has `delete`, but I was thinking this logic may make more sense on the Controller actually.

When the Controller detects a retry, it can check the Workflow's child Pods.

Need to think a bit more about how that would work, but that would preserve the existing separation of duties between Server and Controller, where the Server is just a simple intermediary for users that can be bypassed with correct RBAC.

Reply:
Hmmm I see that `podsToDelete` _already_ broke the separation a bit... we honestly may want to refactor that too...

@terrytangyuan do you have any thoughts on this? Specifically, the current behavior with the Server having Pod modification logic actually invalidates your answer in https://github.com/argoproj/argo-workflows/discussions/12027#discussioncomment-7316157 . I think it ideally _should_ behave the way you described, if possible.

Reply:
You are right. RBAC modifications is required for my answer in https://github.com/argoproj/argo-workflows/discussions/12027#discussioncomment-7316157

Reply:
@terrytangyuan I was actually looking for your thoughts on the approach. I think we should refactor this logic to not have the Server perform anything but a label modification, and then the Controller should actually delete, reset, etc child Pods as needed (as that is the Controller's responsibility, not the Server's)

Reply:
@terrytangyuan  I agree with @agilgur5. Server should not update or delete other than `Argo workflow` CRD.  If some user directly updates the WF spec then this will not work. Controller should handle the scenario to clean the workflow pods based on GCStrategy. 
I remember @ishitasequeira was proposed to delete all workflow pods "label based" like `workflowname` deletion if the workflow complete.

Reply:
Julie agreed in https://github.com/argoproj/argo-workflows/pull/12419#issuecomment-1896588824 as well, so I filed #12538 as a tracking issue for this refactor

Reply:
SGTM